tools/image-audit/scripts.js

```diff
@@ -545,78 +506,180 @@ function getFormData(form) {
+ * @returns {Promise<>} Object with `live` and `preview` hostnames.
+ */
+async function fetchHosts(org, site) {
+  let status;
+  try {
+    const url = `https://admin.hlx.page/status/${org}/${site}/main`;
+    const res = await fetch(url);
+    status = res.status;
+    const json = await res.json();
+    return {
+      status,
+      live: new URL(json.live.url).host,
+      preview: new URL(json.preview.url).host,
+    };
+  } catch (error) {
+    return {
+      status,
+      live: null,
+      preview: null,
+    };
+  }
+}
+
+/* setup */
+async function processForm(sitemapUrls) {
+  // const urls = await fetchSitemap(sitemap);
+  // await fetchAndDisplayBatches(urls.slice(8000, 8100));
+  await fetchAndDisplayBatches(sitemapUrls);
+}
+
 function registerListeners(doc) {
-  const URL_FORM = doc.getElementById('site-form');
-  const CANVAS = doc.getElementById('canvas');
-  const GALLERY = CANVAS.querySelector('.gallery');
-  const DOWNLOAD = doc.getElementById('download-report');
-  const ACTION_BAR = CANVAS.querySelector('.action-bar');
-  const SORT_ACTIONS = ACTION_BAR.querySelectorAll('input[name="sort"]');
-  const FILTER_ACTIONS = ACTION_BAR.querySelectorAll('input[name="filter"]');
+  const form = doc.getElementById('search-form');
+  const error = doc.querySelector('.error');
+  const errorTitle = error.querySelector('.error-title');
+  const errorMsg = error.querySelector('.error-msg');
+  const canvas = doc.getElementById('canvas');
+  const gallery = canvas.querySelector('.gallery');
+  const downloadReport = doc.getElementById('download-report');
+  const actionbar = canvas.querySelector('.action-bar');
+  const sortActions = actionbar.querySelectorAll('input[name="sort"]');
+  const filterActions = actionbar.querySelectorAll('input[name="filter"]');
+
+  /**
+   * Handles admin form submission.
+   * @param {Event} e - Submit event
+   */
 
   // handle form submission
-  URL_FORM.addEventListener('submit', async (e) => {
+  form.addEventListener('submit', async (e) => {
     e.preventDefault();
+    updateConfig();
+    error.setAttribute('aria-hidden', 'true');
+    errorTitle.textContent = 'Error';
+    errorMsg.textContent = '';
     // clear all sorting and filters
     // eslint-disable-next-line no-return-assign
-    [...SORT_ACTIONS, ...FILTER_ACTIONS].forEach((action) => action.checked = false);
-    const data = getFormData(e.srcElement);
-    const url = data['site-url'];
-    const urlType = extractUrlType(url);
-    if (urlType.includes('sitemap')) {
-      // fetch sitemap
-      const sitemap = urlType === 'sitemap' ? url : writeSitemapUrl(url);
-      processForm(sitemap);
+    [...sortActions, ...filterActions].forEach((action) => action.checked = false);
+    const {
+      org, site, sitemap, path,
+    } = getFormData(form);
+    // fetch host config
+    const { status, live } = await fetchHosts(org, site);
+    if (!live || status !== 200) {
+      return;
```

Comment:
I think we should show an error in this case?

Reply:
Also, similar to https://labs.aem.live/tools/site-query/index.html this requires CORS to work properly. You may want to do a similar error message thing as we do there to warn users of that.